src/modules/systemlib/airspeed.c

```diff
@@ -55,59 +55,107 @@
  * @param differential_pressure total_ pressure - static pressure
  * @return indicated airspeed in m/s
  */
-float calc_indicated_airspeed_corrected(enum AIRSPEED_PITOT_MODEL pmodel, enum AIRSPEED_SENSOR_MODEL smodel,
-					float tube_len, float differential_pressure, float pressure_ambient, float temperature_celsius)
+float calc_indicated_airspeed_corrected(enum AIRSPEED_COMPENSATION_MODEL pmodel, enum AIRSPEED_SENSOR_MODEL smodel,
+					float tube_len, float tube_dia_mm, float differential_pressure, float pressure_ambient, float temperature_celsius)
 {
 
 	// air density in kg/m3
-	double rho_air = get_air_density(pressure_ambient, temperature_celsius);
+	const double rho_air = get_air_density(pressure_ambient, temperature_celsius);
 
-	double dp = fabsf(differential_pressure);
-	// additional dp through pitot tube
-	float dp_pitot;
-	float dp_tube;
-	float dv;
+	const float dp = fabsf(differential_pressure);
+	float dp_tot = dp;
+
+	float dv = 0.0f;
 
 	switch (smodel) {
 
 	case AIRSPEED_SENSOR_MODEL_MEMBRANE: {
-			dp_pitot = 0.0f;
-			dp_tube = 0.0f;
-			dv = 0.0f;
+			// do nothing
 		}
 		break;
 
 	case AIRSPEED_SENSOR_MODEL_SDP3X: {
-			// flow through sensor
-			double flow_SDP33 = (300.805 - 300.878 / (0.00344205 * pow(dp, 0.68698) + 1)) * 1.29 / rho_air;
+			// assumes a metal pitot tube with round tip as here: https://drotek.com/shop/2986-large_default/sdp3x-airspeed-sensor-kit-sdp31.jpg
+			// and tubing as provided by px4/drotek (1.5 mm diameter)
+			// The tube_len represents the length of the tubes connecting the pitot to the sensor.
+			switch (pmodel) {
+			case AIRSPEED_COMPENSATION_MODEL_PITOT:
+			case AIRSPEED_COMPENSATION_MODEL_NO_PITOT: {
+					const double dp_corr = (double)dp * 96600.0 / (double)pressure_ambient;
+					// flow through sensor
+					double flow_SDP33 = (300.805 - 300.878 / (0.00344205 * pow(dp_corr, 0.68698) + 1)) * 1.29 / rho_air;
```

Comment:
Do we actually need double precision?

Reply:
I am not sure, but I will check it.

Reply:
I changed it and did a test flight with it. I saw no issue so it should be fine.